staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/meta.go

```diff
@@ -54,6 +54,7 @@ type Object interface {
 	GetLabels() map[string]string
 	SetLabels(labels map[string]string)
 	GetAnnotations() map[string]string
+	GetAnnotationsSafely() (map[string]string, error)
```

Comment:
Looks like the problem is that `Unstructured` claims to implement the `Object` interface but silently fails and returns a `nil` map if `NestedStringMap` return an error. Note that `GetLabels` has the same issue, and whatever we decide to do, we apply the same treatment to it.

I agree that the way `Unstructured#GetAnnotations` silent fails is a problem, but I'm not convinced that expanding the interface is a good path forward. Adding new functions to interfaces is a breaking change since existing implementations must be updated to implement the function to stay compatible.

Instead, could `Unstructured` do whatever data integrity checks it needs to do upfront such that there is no possibility of error during field access? That would seem more in line with the existing interface.

Reply:
I agree an API change is probably not the best solution. Finding an alternative place to do this check is not easy. I've tried to scout around and found [here](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured/helpers.go#L326-L344) and [here](pkg/kubectl/cmd/util/openapi/validation/validation.go). Any other suggestions would be welcomed.

Reply:
How about a `Validate` function on `Unstructured` that is called separately? That gives callers the option to validate but don't make it mandatory.